### PR TITLE
Stop hammering Zenodo for generic URLs when it keeps failing

### PR DESCRIPTION
**Pull request by nemobis**
Avoids this kind of tedious and pointless case where we fail a hundred times in a row:
https://phabricator.wikimedia.org/P8064

### CODE REVIEW

File: api_handlers/zotero.php [hunk 1]
```diff
@@ -10,7 +10,7 @@ function query_url_api($ids, $templates) {
     if ($failures >= 5 ) {
       // Repeat failures for errors on our side: probably a chronic timeout.
       report_action("Giving up on generic URLs for this page.")
-      break;
+      return;
```
Comment (nemobis) [inline api_handlers/zotero.php]: No, this is not what I intended. In my example, the rest of the operation continued correctly. Please don't skip the entire thing.

File: api_handlers/zotero.php [hunk 2]
```diff
@@ -144,7 +142,7 @@ function expand_by_zotero(&$template, $url = NULL) {
       return FALSE;
     case 'Remote page not found':
       report_info("Remote page not found for URL ". $url);
-      return NULL;
+      return FALSE;
   }
   
   if (strpos($zotero_response, '502 Bad Gateway') !== FALSE) {
```
Comment (nemobis) [inline api_handlers/zotero.php]: My idea was to count 502 errors from Tools and the other failures on our side, but ok.

**Comment by GlazerMann** [comment]
The only thing that really points to the tool having trouble is the timeouts

**Comment by GlazerMann** [comment]
@nemobis how about this.  We keep the dropping of urls code and such.

**Comment by nemobis** [comment]
Sure, this could be enough. I'm not sure if the global is going to cause trouble later.

**Pull request merged**